tests/jballerina-unit-test/src/test/resources/test-src/expressions/object/object_closures.bal

```diff
@@ -337,6 +355,369 @@ function testFunctionPointerAsFieldValue() {
+            }
+        }
+    };
+
+    var val = wait callMethod(obj1, "$get$foo");
+    if val is int {
+        var val2 = wait callMethod(obj1, "bar");
+        if (val2 is int) {
+            return val + val2;
+        }
+    }
+    return 0;
+}
+
+type Foo object {
+    int i;
+    function foo(int b2) returns int;
+};
+
+type Bar object {
+    function bar(int b3) returns object {};
+};
+
+function testClosuresWithObjectConstrExpr10(int b1) returns int {
+    final int a1 = 10;
+
+    var obj1 = object Foo {
+        int i;
+
+        function init() {
+            self.i = b1 + a1;
+        }
+
+        function foo(int b2) returns int {
+            Bar obj2 = object {
+                function bar(int b3) returns object {
+                    int i;
+                    function foo(int b4) returns int;
+                } {
+                    return object {
+                        int i = 10;
+                        function foo(int b4) returns int {
+                            return intVal + b4;
+                        }
+                    };
+                }
+            };
+            Foo obj3 = <Foo> obj2.bar(10);
+            return a1 + self.i + b1 + obj3.foo(50);
+        }
+    };
+
+    return obj1.foo(20);
+}
+
+function testClosuresWithObjectConstrExpr11(int b1, string str) returns int {
+    final int a1 = 10;
+
+    var obj1 = object {string j = str; boolean k = false;};
+    Foo[] arr = [
+        object {
+            int i;
+
+            function init() {
+                self.i = b1 + a1;
+            }
+
+            function foo(int b2) returns int {
+                Bar obj2 = object {
+                    function bar(int b3) returns object {} {
+                        return object {string j = str; boolean k = true;};
+                    }
+                };
+                object {string j; boolean k;} obj3 = <object {string j; boolean k;}> obj2.bar(10);
+                if obj1.k != obj3.k {
+                    self.i = 200;
+                }
+                return b1 + a1 + self.i + b2 + intVal;
+            }
+        }
+    ];
+
+    return arr[0].foo(10);
+}
+
+const number = 10;
+
+function testClosuresWithObjectConstrExpr12(int b1) returns int {
+    final int a1 = 100;
+
+    var obj1 = object {
+        function bar(int b3) returns int {
+            if object {
+                int i = number;
+                function foo(int b2) returns int {
+                    return self.i + b1;
+                }
+            } === object {
+                int i = number;
+                function foo(int b2) returns int {
+                    return self.i + b2;
+                }
+            } {
+                return a1;
+            }
+            return b1 + b3 + a1;
+        }
+    };
+
+    return obj1.bar(10);
+}
+
+function testClosuresWithObjectConstrExpr() {
+    assertValueEquality("CACBC", testClosuresWithObjectConstrExpr1("C"));
```

Comment:
We need to assert each function separately otherwise if one fails the rest won't run